File: go/reporter.go
package beval

import "encoding/json"

// ToJSON serializes a RunResult to a JSON byte slice.
func ToJSON(result RunResult, indent bool) ([]byte, error) {
	if indent {
		return json.MarshalIndent(result, "", "  ")
	}
	return json.Marshal(result)
}

// WriteJSON serializes a RunResult and writes it to the given path.
//
// Uses os.WriteFile internally; the caller is responsible for ensuring
// the output directory exists.
func WriteJSON(result RunResult, path string) error {
	data, err := ToJSON(result, true)
	if err != nil {
		return err
	}
	// Import os at usage site to keep stub minimal.
	// Will use: os.WriteFile(path, data, 0o644)
	_ = data
	_ = path
	return nil
}

File: typescript/src/reporter.ts
/**
 * Result reporting for the beval framework.
 *
 * Formats and outputs evaluation results to console or JSON.
 * See SPEC.md §12 (Results Schema).
 */

import type { RunResult } from "./types.js";

/** Serialize a RunResult to a JSON string. */
export function toJson(result: RunResult, indent = 2): string {
  return JSON.stringify(result, null, indent);
}

/** Write a RunResult to a JSON file. */
export async function writeJson(result: RunResult, path: string): Promise<void> {
  const { writeFile } = await import("node:fs/promises");
  await writeFile(path, toJson(result), "utf-8");
}
